### PR TITLE
rgw: cors rules num limit

### PR DESCRIPTION
**Pull request by joke-lee**
According to AWS S3 (https://docs.aws.amazon.com/AmazonS3/latest/dev/cors.html), an cors rules can
have up to 100 rules.

Signed-off-by: yuliyang <yuliyang@cmss.chinamobile.com>

### CODE REVIEW

**Comment by joke-lee** [comment]
hi @cbodley , do you mind take a review? thanks

**Comment by joke-lee** [comment]
s3tests https://github.com/ceph/s3-tests/pull/240

**Comment by joke-lee** [comment]
jenkins test this please

File: src/common/legacy_config_opts.h [hunk 1]
```diff
@@ -1540,3 +1540,4 @@ OPTION(rgw_max_objs_per_shard, OPT_INT)
 OPTION(rgw_reshard_thread_interval, OPT_U32) // maximum time between rounds of reshard thread processing
 
 OPTION(rgw_acl_grants_max_num, OPT_INT) // According to AWS S3(http://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html), An ACL can have up to 100 grants.
+OPTION(rgw_cors_rules_max_num, OPT_INT) // According to AWS S3(http://docs.aws.amazon.com/AmazonS3/latest/dev/cors.html), An cors can have up to 100 rules.
```
Comment (mattbenjamin) [inline src/common/legacy_config_opts.h]: ok, a configurable with default same as AWS seems fine

File: src/rgw/rgw_rest_s3.cc [hunk 2]
```diff
@@ -2379,6 +2379,23 @@ int RGWPutCORS_ObjStore_S3::get_params()
     return -EINVAL;
   }
 
+#define CORS_RULES_MAX_NUM      100
```
Comment (mattbenjamin) [inline src/rgw/rgw_rest_s3.cc]: ok

**Comment by joke-lee** [comment]
hi @yuriw can you help to add this to qa test , thanks.

**Pull request merged**